File: task1/pages/homepage.py
from selenium.webdriver.common.by import By

class HomePage:
    #Finding the link to the login page
    LOGIN_LINK = (By.XPATH, './/a[@href="/login"]')
    HOMEPAGE_LINK = "https://automationexercise.com"

    def __init__(self, driver):
        self.driver = driver
    #Clicking the link to the login page
    def go_to_login_page(self):
        login_link = self.driver.find_element(*self.LOGIN_LINK)
        login_link.click()
    #Simply navigating to the homepage
    def navigate(self):
        self.driver.get(self.HOMEPAGE_LINK)

File: CandTS/task1/cypress/support/pages/HomePage.ts
/// <reference types="Cypress" />

export default class HomePage {
    LOGIN_LINK = "a[href='/login']";
    HOMEPAGE_LINK = "https://automationexercise.com/";

    //Simply navigating to the homepage
    navigate() {
        return cy.visit(this.HOMEPAGE_LINK).then(() => {
            cy.url().should("eq", this.HOMEPAGE_LINK);
        });
    }

    //Clicking the link to the login page
    goToLoginPage() {
        cy.get(this.LOGIN_LINK).click();
    }
}
